Features.md

```diff
@@ -0,0 +1,140 @@
+# Reddit Clone
+
+ACCOUNT
+
+Signup with email:
+- takes an email, password and an unique username 
+- creates a new user in the db and stores the password in a hash format
+- sends out access and refresh token(JWT) as response
+
+Signup with Social(gmail):
+- allows anyone to create a new account using their gmail account
+- generates a random username internally for this user 
+
+Login:
+- takes username and password as input
+- if they match an entry in the users model, then send access and refresh token(JWT) as response
+
+Login with Social(gmail):
+- allows logging in with a gmail account
+- create a new user if user not present already i.e., generate a random username
+
+Forgot username:
+- login needs username and password in reddit
+- if username is forgotten, we can use this feature to send out an email to the registered email id with the matching username
+- takes an email id as input, sends an email to the same if it matches with an entry in users model, else throws error
+
+Forgot password:
+- this can be used to reset a password when a password is forgotten, to send out an email with the reset link 
+- takes username and email as input, checks if it matches an entry in the users model, if it does, then sends an email with a password reset link,
+  else throws error
+- the password link itself will have a limited lifespan and once used cannot be used again
+
+List all users:
+- creates a master list of sorts for the user model i.e., lists all the users signed up
+- gives a paginated response
+
+List followed users:
```

Comment:
Don't see any functionality that implements following/unfollowing a user

Reply:
added it under general